### PR TITLE
source-oracle: skip archived log if it has same sequence as current log

### PR DESCRIPTION
**Pull request by mdibaiee**
**Description:**

- Apparently it is possible for the current log file to be at the same time archived, i.e. it is possible to have two log files with the same `SEQUENCE#` and `FIRST_CHANGE#`, one of them with `Status: ARCHIVED` and the other `Status: CURRENT`. In this scenario Logminer complains:
```
adding logfile "/rdsdbdata/db/OOPUAE_A/onlinelog/o1_mf_3_m5ml99to_.log" (CURRENT, 18433725214, NO, NO) to logminer: ORA-01289: cannot add duplicate logfile /rdsdbdata/db/OOPUAE_A/onlinelog/o1_mf_3_m5ml99to_.log
ORA-06512: at "SYS.DBMS_LOGMNR", line 82
ORA-06512: at line 1
```

This pull-request addreses this case by preferring the `CURRENT` file over the archived one (with the presumption that the `CURRENT` file has more recent updates). Added a debug log for this scenario so I can monitor it. I had not seen this case in my tests ever, but just saw it with a customer.

I have run our test suite on this but we don't cover this edge case, I am not sure how to reproduce this scenario. I plan to first roll this out to that one customer to make sure it works correctly before merging, but need an approval for that.

**Workflow steps:**

(How does one use this feature, and how has it changed)

**Documentation links affected:**

(list any [documentation links](https://docs.google.com/document/d/1SRC9VS9zyCzWl3n4HXHbc4wPB1eLxJHkA2rtu9ZNokM/edit?usp=sharing) that you created, or existing ones that you've identified as needing updates, along with a brief description)

**Notes for reviewers:**

(anything that might help someone review this PR)

- - -
This change is [Reviewable](https://reviewable.io/reviews/estuary/connectors/2531)

### CODE REVIEW

File: source-oracle/replication.go [hunk 1]
```diff
@@ -193,17 +193,42 @@ func (s *replicationStream) addLogFiles(ctx context.Context, startSCN, endSCN in
...
 		redoFiles = append(redoFiles, f)
 
-		// once we hit a log file that has passed endSCN and it signifies a dictEnd, we know we don't
-		// need any more log files. If we don't include a dictionary end file, we risk having an incomplete
-		// dictionary
-		if f.FirstChange >= endSCN && f.DictEnd == "YES" {
-			break
+		if f.FirstChange >= endSCN {
+			if s.db.config.Advanced.DictionaryMode == DictionaryModeOnline {
+				break
```
Comment (mdibaiee) [inline source-oracle/replication.go:224]: In online mode we don't need to add all files until dictionary end, as soon as we have covered the SCN range, we are good
